src/components/engagement/pnp-extractor.service.ts

```diff
@@ -16,12 +16,47 @@ export class PnpExtractor {
     @Logger('pnp:extractor') private readonly logger: ILogger
   ) {}
 
+  private year = 0;
+  private quarter = 0;
```

Comment:
Class properties to store temporary data is a bad practice and hard to follow. Refactor so that it's returned from this method instead. The `extract()` method could be refactored to easily join the results of this and the excel data into a single object.